### PR TITLE
Vantiv Certify Script with XML

### PR DESCRIPTION
**Pull request by scottbartell**
add flag use_xml flag to vantiv certify script and get it workin…g with direct tokenization certifications

### CODE REVIEW

File: lib/vantiv/certification/validation_test_runner.rb [hunk 1]
```diff
@@ -89,20 +91,28 @@ def shutdown
...
+          response.body.authorization_response ||
+          response.body.sale_response ||
+          response.body.credit_response ||
+          response.body.void_response ||
+          response.body.auth_reversal_response ||
+          response.body.capture_response
 
         response_cache.push(cert_name, response)
-        results_file << "#{cert_name},#{response.body.request_id}\n"
+        results_file << "#{cert_name},#{transaction_response.transaction_id}\n"
```
Comment (scottbartell) [inline lib/vantiv/certification/validation_test_runner.rb]: This needs to change - it breaks non-xml certifications.

File: lib/vantiv/certification/validation_test_runner.rb [hunk 2]
```diff
@@ -89,20 +91,28 @@ def shutdown
...
-          response_object: Vantiv::Api::Response.new
+          response_object: Vantiv::Api::Response.new,
+          use_xml: use_xml
         ).run
 
         if response.api_level_failure?
           error_message = "CERT FAILED: #{cert_name} \n WITH: #{response.body}"
           raise StandardError.new(error_message)
         end
+        transaction_response = response.body.register_token_response ||
```
Comment (scottbartell) [inline lib/vantiv/certification/validation_test_runner.rb]: I would like to figure out a cleaner way to do this.

File: exe/vantiv-certify-app [hunk 3]
```diff
@@ -31,6 +31,10 @@ parser = OptionParser.new do |opts|
   opts.on("--filter_by [TEST_NAME]", "Filter which certifications to run") do |filter|
     options[:filter_by] = filter
   end
+
+  opts.on("--use_xml", "Use XML endpoint") do |use_xml|
```
Comment (ChrisVoxland) [inline exe/vantiv-certify-app]: this seems like it should just be true if the flag is there, and false otherwise

Comment (scottbartell) [inline exe/vantiv-certify-app]: This is a boolean flag and doesn't actually take in a value. So if the flag is present then the value is actually set to the boolean value `true`.

File: lib/vantiv/certification/validation_test_runner.rb [hunk 4]
```diff
@@ -89,20 +101,33 @@ def shutdown
...
         end
 
+        transaction_id = get_transaction_id(response, use_xml)
+
         response_cache.push(cert_name, response)
-        results_file << "#{cert_name},#{response.body.request_id}\n"
+        results_file << "#{cert_name},#{transaction_id}\n"
+      end
+
+      def get_transaction_id(response, use_xml)
```
Comment (ChrisVoxland) [inline lib/vantiv/certification/validation_test_runner.rb]: don't need argument

File: lib/vantiv/certification/validation_test_runner.rb [hunk 5]
```diff
@@ -89,20 +101,33 @@ def shutdown
...
+          use_xml: use_xml
         ).run
 
         if response.api_level_failure?
-          error_message = "CERT FAILED: #{cert_name} \n WITH: #{response.body}"
+          error_message = "CERT FAILED: #{cert_name} \n WITH: #{response.raw_body}"
           raise StandardError.new(error_message)
         end
 
+        transaction_id = get_transaction_id(response, use_xml)
```
Comment (ChrisVoxland) [inline lib/vantiv/certification/validation_test_runner.rb]: don't need use_xml argument

**Pull request merged**